homeassistant/components/nest/quality_scale.yaml

```diff
@@ -0,0 +1,77 @@
+rules:
+  # Bronze
+  config-flow:
+    status: todo
+    comment: Some fields are missing a data_description
+  brands: done
+  dependency-transparency: done
+  common-modules:
+    status: exempt
+    comment: The integration does not have a base entity or coordinator.
+  has-entity-name: done
+  action-setup:
+    status: exempt
+    comment: The integration does not register actions.
+  appropriate-polling:
+    status: exempt
+    comment: The integration does not poll.
+  test-before-configure:
+    status: todo
+    comment: |
+      The integration does a connection test in the configuration flow, however
+      it does not fail if the user has ipv6 misconfigured.
+  entity-event-setup: done
+  unique-config-entry: done
+  entity-unique-id: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  test-before-setup:
+    status: todo
+    comment: |
+      The integration does tests on setup, however the most common issues
+      observed are related to ipv6 misconfigurations and the error messages
+      are not self explanatory and can be improved.
+  docs-high-level-description: done
+  config-flow-test-coverage: done
```

Comment:
Is it possible for the user to reauth with a different account?

Reply:
I believe the [original choice](https://github.com/home-assistant/core/pull/73720) to use device access project id was because the nest API does not return any notion of identity.    The device access project is currently the unique id and we've asked users to set this up to create their own project id, but theoretically you could share them across accounts.

In years following, the pattern that newer google integrations have followed is to additionally query the user identify API to get the account information. We could follow here also and enforce going forward.

Reply:
I added this as a comment for re-authentication item in the silver category (didn't yet grade this yet)

Reply:
Of course -- all changes come with tests :)